File: Ruby/2E.rb
# Input: The input cosists of a single string, consisting of characters and numbers, the string's length doesn't exceed 200 characters.
# Output: Return an integer by counting the total duplicate values

# Example: "abcde" => 0
# Example: "aabbcde" => 2
# Example: "aabBcde" => 2
# Example: "indivisibility" => 1
# Example: "Indivisibilities" => 2
# Example: "aA11" => 2
# Example: "ABBA" => 2

def duplicateCount (input)
    num = 0
	input = input.downcase
	chars = input.split('')
    
    chars.each do|c|
      num = input.scan(c).count
    	end
	return num
end

File: NodeJS/2E.js
/* 
    Input: The input consists of a single string, consisting of characters and numbers, the string's length doesn't exceed 200 characters.
    Output: Return the string by masking all but the last 4 digits.
    
    Example maskify('4556364607935616') => '############5616'
    Example maskify('64607935616') => '#######5616'
    Example maskify('1') => '1'
    Example maskify('') => ''
*/

const maskify = (input) => {
    return input.replace(/.(?=.{4})/g, '#');
}